test/service/FileTransferService.test.ts

```diff
@@ -79,4 +80,67 @@ describe('FileTransferService', () => {
             }
         });
     });
+
+    describe('delete file', () => {
+        it('should throw an error when file ID is not defined', () => {
+
+            [undefined, null].forEach(async evidence => {
+                try {
+                    await fileTransferService.delete(evidence!)
+                } catch (err) {
+                    expect(err).to.be.instanceOf(Error)
+                        .and.to.haveOwnProperty('message').equal('File ID is missing')
+                }
+            })
+        });
+
+        it('should throw an error when file does not exist', async () => {
+
+            const fileId = 'non-existing-file';
+
+            nock(HOST)
+                .delete(`${URI}/${fileId}`)
+                .reply(404, {
+                    message: 'file not found'
+                });
+
+            try {
+                await fileTransferService.delete(fileId);
+                assert.fail('Test should failed while it did not')
+            } catch (err) {
+                expect(err.message).to.contain(`File ${fileId} cannot be deleted because it does not exist`);
```

Comment:
can we 'expect' the error code here too?

Reply:
Error throw by an API does not come with error code. It hides response code concern from service consumers.